lib/DDG/Meta/Data.pm

```diff
@@ -185,4 +201,36 @@ sub _js_callback_name {
     return $fn;
 }
 
+# return fathead data
+sub source_data {
+    my ($self, $source) = @_;
+    if(my $data = $ia_metadata{source}{$source}){
+        return (
```

Comment:
Can the code using these functions not handle an empty array?  This really should be:
``` Perl
my @d;
if(my $ia = ...){
    @d = (
    );
}
return @d;
```
This empty string stuff shouldn't be necessary, unless you're getting undefined errors?